bdai_ros2_wrappers/bdai_ros2_wrappers/tf_listener_wrapper.py

```diff
@@ -1,154 +1,120 @@
 # Copyright (c) 2023 Boston Dynamics AI Institute Inc.  All rights reserved.
-import time
-from typing import Optional, Tuple
 
-import rclpy
+from typing import Optional
+
 from geometry_msgs.msg import TransformStamped
+from rclpy.clock import ClockType
 from rclpy.node import Node
+from rclpy.task import Future
 from rclpy.time import Duration, Time
-from tf2_ros import ExtrapolationException, TransformException
+from tf2_ros import TransformException
 from tf2_ros.buffer import Buffer
 from tf2_ros.transform_listener import TransformListener
 
 import bdai_ros2_wrappers.scope as scope
+from bdai_ros2_wrappers.futures import wait_for_future
 
 
-class TFListenerWrapper(object):
-    def __init__(
-        self,
-        node: Optional[Node] = None,
-        wait_for_transform: Optional[Tuple[str, str]] = None,
-        cache_time_s: Optional[int] = None,
-    ) -> None:
+class TFListenerWrapper:
+    def __init__(self, node: Optional[Node] = None, cache_time_s: Optional[float] = None) -> None:
```

Comment:
What happeneed to the wait for init part?  I do know people were using that...

Reply:
Yes, quite a bit in fact (and not always correctly e.g. [here we pass three frames](https://github.com/bdaiinstitute/bdai/blob/4e336b375747c2bcd7a20da045b165bbed46fc9e/ws/src/spot_utilities/spot_utilities/spot_grasping_server.py#L37)). It is a problematic pattern, though. When the wrapper is bound to a node that is spinning, all is fine with waiting on a constructor. When the wrapper gets instantiated from within the constructor of the node it is bound to, it hangs, because the node can't be put to spin until it's fully constructed. 

Eliminating that pattern was the original reason behind this PR. You can still wait, but you have do to so explicitly after construction.

Reply:
OK.  Could we add a comment here explaining how to do that?  The reason we added this was because people were constantly getting exceptions thrown from trying to use the listener too soon.

Reply:
Done in https://github.com/bdaiinstitute/ros_utilities/pull/35/commits/fc71daa4b76f7c854240665a62f54527f620d797. I added documentation, including basic examples, and it will also now generate warnings if you are about to shoot yourself in the foot. PTAL!